src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -256,43 +269,56 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
             }
         }
 
-        Map<String, GoogleBucketResource> bucketCache = new HashMap<>();
+        Map<String, GoogleBucketResource> bucketCache = new ConcurrentHashMap<>();
```

Comment:
Maybe comment what String will be holding in this map

Reply:
👍